### PR TITLE
OCPBUGS-31626: Stop checking CloudControllerOwner

### PR DESCRIPTION
**Pull request by sebsoto**
This commit reacts to
https://github.com/openshift/cluster-cloud-controller-manager-operator/commit/65c07a6e44bfa1b0cdfadc264d3341c093ffd914

There is no longer a CloudControllerOwner condition on the CCM clusteroperator, as external cloud controllers are now fully default. WMCO will no longer check for this condition, and the logic depending on it should now always be done.

### CODE REVIEW

**Review by saifshaikh48 — COMMENTED** [review]
/lgtm

:rocket:

**Review by jrvaldes — APPROVED** [review]
/lgtm

**Comment by sebsoto** [comment]
/approve

Self approving now that multiple people have `/lgtm`'d, in order to unblock merges to master.

**Pull request merged**
